StripeConnect/StripeConnect/Source/Internal/Webview/ConnectWebView.swift

```diff
@@ -158,27 +178,128 @@ extension ConnectWebView: WKNavigationDelegate {
     ) async -> WKNavigationResponsePolicy {
-        // TODO: MXMOBILE-2485 Handle downloads
-        .allow
+
+        // Downloads will typically originate from a non-allow-listed host (e.g. S3)
+        // so first check if the response is a download before evaluating the host
+
+        // The response should be a download if its Content-Disposition is
+        // shaped like `attachment; filename=payouts.csv`
+        if navigationResponse.canShowMIMEType,
+           let response = navigationResponse.response as? HTTPURLResponse,
+           let contentDisposition = response.value(forHTTPHeaderField: "Content-Disposition"),
+           contentDisposition
+            .split(separator: ";")
+            .map({ $0.trimmingCharacters(in: .whitespaces) })
+            .caseInsensitiveContains("attachment") {
+            return .download
+        }
+
+        return .allow
     }
 
     func webView(_ webView: WKWebView,
                  navigationAction: WKNavigationAction,
                  didBecome download: WKDownload) {
-        // TODO: MXMOBILE-2485 Handle downloads
+        download.delegate = self
     }
 
     func webView(_ webView: WKWebView,
                  navigationResponse: WKNavigationResponse,
                  didBecome download: WKDownload) {
-        // TODO: MXMOBILE-2485 Handle downloads
+        download.delegate = self
+    }
+}
+
+// MARK: - WKDownloadDelegate implementation
+
+@available(iOS 15, *)
+extension ConnectWebView {
+    // This extension is an abstraction layer to implement `WKDownloadDelegate`
+    // functionality and make it testable. There's no way to instantiate
+    // `WKDownload` in tests without causing an EXC_BAD_ACCESS error.
+
+    func download(decideDestinationUsing response: URLResponse,
+                  suggestedFilename: String) async -> URL? {
+        // The temporary filename must be unique or the download will fail.
+        // To ensure uniqueness, append a UUID to the directory path in case a
+        // file with the same name was already downloaded from this app.
+        let tempDir = fileManager
+            .temporaryDirectory
+            .appendingPathComponent(UUID().uuidString)
+        do {
+            try fileManager.createDirectory(at: tempDir, withIntermediateDirectories: true)
+        } catch {
+            showErrorAlert(for: error)
+            return nil
+        }
+
+        downloadedFile = tempDir.appendingPathComponent(suggestedFilename)
+        return downloadedFile
+    }
+
+    func download(didFailWithError error: any Error,
+                  resumeData: Data?) {
+        showErrorAlert(for: error)
+    }
+
+    func downloadDidFinish() {
+
+        // Display a preview of the file to the user
+        let previewController = QLPreviewController()
+        previewController.dataSource = self
+        previewController.modalPresentationStyle = .pageSheet
+        presentPopup(previewController)
```

Comment:
Would we want to just present the share sheet directly here rather than having to fix potential bugs involving previewing certain file types?

Reply:
We could – let me get feedback from the team what the preference is.

Reply:
Looks like we can check if the file is previewable, and if it's not then directly open the share sheet.
https://developer.apple.com/documentation/quicklook/qlpreviewcontroller/canpreview(_:)

Really good call out – from the docs:
> [Discussion](https://developer.apple.com/documentation/quicklook/qlpreviewcontroller/canpreview(_:)#Discussion)
> If the system can’t display an item, but you still attempt to display it, a Quick Look preview controller displays a generic error. Always check whether you can display an item before choosing to do so.

Reply:
I'm going to merge this but do a fast-follow PR to address this issue

Reply:
https://jira.corp.stripe.com/browse/MXMOBILE-2827